### ISSUE TITLE
Controls failing with some RTMP streams

### ISSUE THREAD

**Issue by ghost** [videojs/video-js-swf#116]
Some RTMP streams return json data that fails to parse. The video is streaming properly until i.e. pause button is pressed, which stops the playback and breaks the player, so that it can't restart (a page refresh is needed).

Debug consoles of various browsers throw errors like:

```
FF: "SyntaxError: missing : after property id"
Chrome: "Uncaught SyntaxError: Unexpected token -"
IE: "Expected ':' char"
```

Here's a test case: http://jsfiddle.net/s8hC3/

Same issue was extensively covered here: http://www.longtailvideo.com/support/forums/jw-player/servers-and-streaming/19772/javascript-error-missing-after-property-id/
Unfortunately it's gone now and I can't find it anymore. Long story short: the flash player doesn't seem to like params with "-" char, maybe others aswell.

**Comment by marclipovsky** [comment]
Having the same issue. Any solution for this yet?

**Comment by mymainmanbrown** [comment]
This appears to still be an issue, unless a more recent build fixes it.  I will try that next, but the version that is available from the CDN - http://vjs.zencdn.net/swf/5.3.0/video-js.swf - still has this issue.

As mentioned in the issue on the videojs repo - https://github.com/videojs/video.js/issues/1071 - this results in not being able to play the stream after it is paused.  There may be other issues, but I have not yet run into them.

Also, even with autoplay set to true and calling this.play in the onReady callback, only one stream will actually autoplay at a time.  I'm not sure if these two things are related.

**Comment by mymainmanbrown** [comment]
I was unable to fix the problem, even after using the 5.4.0 build from the CDN:

```
this.player = window.videojs(this.videoNode, {
  autoplay: true,
  muted: true,
  controls: true,
  preload: 'auto',
  sources,
  flash: {
    swf: 'http://vjs.zencdn.net/swf/5.4.0/video-js.swf',
  },
  techOrder: [
    'flash',
    'html5',
  ],
}, function () {
  this.play();
});
```

I took a look at the error in the chrome console, and here is the offending javascript:

```
__flash__toXML(videojs.Flash.onEvent("vjs_video_724_Flash_api", "loadedmetadata", ({
    displayWidth: 426,
    height: 240,
    trackinfo: [({
        profile-level-id: "MY_ID",
        language: "eng",
        type: "video",
        sampledescription: [({
            sampletype: "H264"
        })],
        timescale: 90000,
        description: "{H264CodecConfigInfo: codec:H264, profile:Baseline, level:3.0, frameSize:320x240, displaySize:426x240, frameRate:15.0, PAR:4:3}",
        sprop-parameter-sets: "some random comma delimited stuff here"
    })],
    width: 426,
    frameWidth: 320,
    videodatarate: 192,
    videocodecid: "avc1",
    framerate: 15,
    frameHeight: 240,
    displayHeight: 240,
    rtpsessioninfo: ({
        name: "Unnamed",
        timing: "0 0",
        protocolversion: "0",
        information: "N/A",
        attributes: ({
            charset: "UTF-8",
            type: "broadcast",
            tool: "MY_TOOL"
        }),
        origin: "MY_ORIGIN",
        connectiondata: "IN IP4 [MY_IP_DATA]"
    })
})));
```

The syntax error occurs because `profile-level-id` and `sprop-parameter-sets` are object keys that are not contained within quotes.  I assume this could be easily fixed, but I have no idea where in the code or in the built swf this information is handled.

The only reference I could find to `videojs.Flash.onEvent` was here - https://github.com/videojs/video-js-swf/blob/8e55ad1eac05dcb6bf8e93d5e8a759f7411a1958/tests/manual/manual.html#L248 - but that doesn't appear to be source code.

I did a little testing with the offending code block, and when I enclose the object properties in strings, all it does is return an xml object of `<undefined/>`, which I assume is not correct.  After looking at the linked "source code" file above, I realized that onEvent doesn't actually take a 3rd parameter, so from what I can tell, that object is never even used.

That's the extent of the information I was able to gather about this for now.  I wasn't able to fix the problem, and had to revert back to jwplayer.

**Comment by juniorh2o** [comment]
I was able to fix the error by parsing the object at line 604

https://github.com/videojs/video-js-swf/blob/8e55ad1eac05dcb6bf8e93d5e8a759f7411a1958/src/com/videojs/providers/RTMPVideoProvider.as#L604

like this:

_model.broadcastEventExternally(ExternalEventName.ON_METADATA, JSON.stringify(_metadata)); 

But it doesn`t fix the control problems
